### PR TITLE
(feat) : initial work to bring HIE workflow to check form

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

### Done
- [x] Add ability to search HEI with mocked function taking `national-id` as the policy-number
### Pending
- [ ] Add ability to check if SHA number exists in HEI
- [ ] Pull the benefits packages and store the values in visit attributes

## Screenshots
https://github.com/user-attachments/assets/6db26e56-09ef-4680-816c-e99e4a4e6d74
## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks, @donaldkibet. Unfortunately, my browser can't play the attached video. You can share it directly on slack.

File: packages/esm-billing-app/src/billing-form/hei-resource.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { FHIRErrorResponse, FHIRPatientResponse } from '../types';
+// TODO: Remove the hardcoded URL and use the correct URL from the environment variables
+const heiUrl = `https://sandbox.tiberbu.health/api/v4/Patient?national-id=`;
```
[the remark this answers is not in the document]
Comment (makombe) [reply packages/esm-billing-app/src/billing-form/hei-resource.tsx:3]: @donaldkibet the abbreviation is it hei or should be hie( To mean health information exchange)?

File: packages/esm-billing-app/src/billing-form/visit-attributes/visit-attributes-form.component.tsx [hunk 2]
```diff
@@ -171,12 +145,14 @@ const VisitAttributesForm: React.FC<VisitAttributesFormProps> = ({
...
                     className={styles.sectionField}
-                    onChange={(e) => field.onChange(e.target.value)}
+                    onChange={({ selectedItem }) => field.onChange(selectedItem)}
                     id="insurance-scheme"
-                    type="text"
-                    labelText={t('insuranceScheme', 'Insurance scheme')}
+                    items={insuranceSchemes}
+                    itemToString={(item) => (item ? item : '')}
+                    titleText={t('insuranceScheme', 'Insurance scheme')}
+                    placeholder={t('selectInsuranceScheme', 'Select Insurance scheme')}
```
Comment (makombe) [inline packages/esm-billing-app/src/billing-form/visit-attributes/visit-attributes-form.component.tsx:155]:
```suggestion
                    placeholder={t('selectInsuranceScheme', 'Select insurance scheme')}
```

**Review by makombe — DISMISSED** [review]
LGTM

**Comment by donaldkibet** [comment]
Kapture 2024-08-19 at 19 14 03

**Pull request merged**